2.0.0

ekmillard opened the pull request:
**SDK Enhancements Summary:**

**Type Annotations & Serializer Support:** Added detailed JMS Serializer annotations for all properties in Account, Dids, and Responses models to improve data handling.

**Error Handling Improvements:** Updated ErrorResponse model to include an optional statusCode, providing more robust error management across SDK calls.

**Method Refactoring:** Refined method names for clarity and consistency (e.g., renamed countryList2 to getRateZone2CountryList).

**Documentation Updates:** Added comprehensive PHPDoc comments to all classes and methods, ensuring clear and detailed documentation for future reference.

**Enhanced Query Handling:** Updated query calls to handle structured error responses and support for optional status codes.

**DidModel Updates:** Implemented fluent setters and added type annotations for all properties, ensuring stronger type safety.

jwilliamsonQB committed:
remove curl wrapper as its no longer neccesary

jwilliamsonQB committed:
add exceptions

jwilliamsonQB committed:
add serializer

jwilliamsonQB committed:
get rid of exceptions

jwilliamsonQB committed:
ignore local test file

jwilliamsonQB committed:
add dedicated class for Error Responses

jwilliamsonQB committed:
dedicated class for Account Detail endpoint

jwilliamsonQB committed:
add serializer and error response

jwilliamsonQB committed:
update getAccountDetails endpoint to return object class

jwilliamsonQB committed:
small refactoring to clean up duplication

jwilliamsonQB committed:
add error response and bool response to set low balance alert

jwilliamsonQB committed:
fix spacing

jwilliamsonQB committed:
update setDailyBalanceAlert

jwilliamsonQB committed:
bug fix

jwilliamsonQB committed:
add objects for getAccountBalance

jwilliamsonQB committed:
update setAutorefill with objects

jwilliamsonQB committed:
add method to reduce duplication

jwilliamsonQB committed:
reduce duplication

jwilliamsonQB committed:
additional models

jwilliamsonQB committed:
add object type hints to all functions

jwilliamsonQB committed:
reorganize models

jwilliamsonQB committed:
reorganize models

jwilliamsonQB committed:
No longer relevant to API as of 2.0.0

jwilliamsonQB committed:
methods are no longer relevant as qubev1 has been made obsolete

jwilliamsonQB committed:
correct incorrect order of directories

CoderJoeW commented on composer.json:
```diff
@@ -10,6 +10,11 @@
+        "jms/serializer": "^3.31.1"
+    },
+    "require-dev": {
+        "phpunit/phpunit": "10.5.x-dev",
```
We should update this to 11.4.3

ekmillard replied:
I agree with this, will update.

CoderJoeW commented on composer.json:
```diff
@@ -10,6 +10,11 @@
+    },
+    "require-dev": {
+        "phpunit/phpunit": "10.5.x-dev",
+        "mockery/mockery": "2.0.x-dev"
```
We should update this to 1.6.12

ekmillard replied:
I agree with this.

CoderJoeW commented on src/History.php:
```diff
@@ -2,35 +2,77 @@
+        int $page = 1,
+        int $perPage = 5,
+        ?string $timezone = null
+    ): string|ErrorResponse {
```
We should probably introduce a model so we dont have such a long parameter list

CoderJoeW commented on src/History.php:
```diff
@@ -2,35 +2,77 @@
-            'per_page'          => $perPage   // 5 - 500
+            'per_page'          => $perPage,
+            'timezone'          => $timezone,
         ];
```
Probably there is a better way then this perhaps we could have some object to array conversion that would reduce the duplication

CoderJoeW commented on src/Ifaxenterprise.php:
```diff
@@ -2,277 +2,449 @@
+     * @param string $isAdmin Flag to set as admin (default: 'off').
+     * @return string|ErrorResponse
+     */
+    public function createUser(string $faxLogin, string $faxPassword, string $sname, string $faxName, ?string $faxLname = null, ?string $faxEmail = null, string $isAdmin = 'off'): string|ErrorResponse
```
Huge parameter list here that could probably be replaced with a model

CoderJoeW commented on src/Ifaxenterprise.php:
```diff
@@ -2,277 +2,449 @@
+     * @param bool $allowListOut Permission to list outgoing faxes.
+     * @return string|ErrorResponse
+     */
+    public function updateUserPermissions(string $faxLogin, string $did, bool $allowSend, bool $allowDelete, bool $allowListIn, bool $allowListOut): string|ErrorResponse
```
Long parameter list

CoderJoeW commented on src/Ifaxenterprise.php:
```diff
@@ -2,277 +2,449 @@
+     * @param string|null $period Optional period filter.
+     * @return string|ErrorResponse
+     */
+    public function faxHistory(?string $did = null, ?string $service = null, ?string $type = null, ?string $faxId = null, ?string $period = null): string|ErrorResponse
```
Long parameter list

CoderJoeW commented on src/Ifaxpro.php:
```diff
@@ -17,21 +36,62 @@ public function listRateCenters($state)
+     * @param string|null $reportAtt Optional report attachment option.
+     * @return string|ErrorResponse|bool Returns true on success, or an error message on failure.
+     */
+    public function orderDid(
```
Long parameter list

Commit:
feat: Add enhanced type annotations, serializer support, and refactor method names across SDK

- Added JMS Serializer annotations for all model properties in the Account, Dids, and Responses models for better serialization handling
- Updated `ErrorResponse` model to include optional `statusCode` for enhanced error handling
- Refactored methods in `Account`, `Dids`, and other classes to align with improved naming conventions (e.g., renamed `countryList2` to `getRateZone2CountryList`)
- Implemented comprehensive PHPDoc comments for all models and methods, ensuring better documentation and clarity
- Adjusted query method calls in classes to support structured error responses and optional status codes
- Enhanced `DidModel` with fluent setters and added missing property annotations for complete type safety

Commit:
chore: update composer dependencies

- Updated `phpunit/phpunit` to version 11.4.3 for enhanced testing capabilities.
- Updated `mockery/mockery` to version 1.6.12 for improved mock testing support.

Commit:
feat: Completed All API integration with response handling and unit tests

- Added missing response models for All APIs to match updated API response schema.
- Refactored response classes to ensure correct casing and data structure alignment:
  - Ensured all response properties use snake_case to align with API fields.
  - Updated models to avoid uninitialized property errors by restructuring fields and adding necessary response mappings.
- Introduced comprehensive unit tests for All API classes and methods:
  - Ensured tests include assertions for no uninitialized properties, correct response parsing, and error handling.
  - Added mock data to simulate API responses for robust testing.
- Introduced a new trait for recursive uninitialized property checks to improve test coverage on nested response structures.

fix: Added helper functions for recursive reflection-based property validation, ensuring deeper structures are fully initialized.

This update finalizes the core API integration and improves error handling and test reliability, ensuring alignment with the latest API specifications.
